File: Q14/Q14.c
/*Max and min*/
#include <stdio.h>
void main()
{
    int n, i, max, min;

    printf("Enter the number of elements in the array: ");
    scanf("%d", &n);

    int arr[n];

    printf("Enter the elements of the array:\n");

    for (i = 0; i < n; i++) 
    {
        scanf("%d", &arr[i]);
    }

    max = arr[0];
    min = arr[0];

    for (i = 1; i < n; i++) 
    {
        if (arr[i] > max)
        {
            max = arr[i];
        }
        if (arr[i] < min)
         {
            min = arr[i];
         }    
    }

    printf("Maximum: %d\n", max);
    printf("Minimum: %d\n", min);

}

File: Q14/Q14.ts
// Find the maximum and minimum numbers in an array.

function findMaxMinTS(arr: number[]): { max: number; min: number } {
  if (arr.length === 0) throw new Error("Array is empty");

  let max = arr[0];
  let min = arr[0];

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] > max) {
      max = arr[i];
    }
    if (arr[i] < min) {
      min = arr[i];
    }
  }
  return { max, min };
}

const numbers = [10, 45, 32, 67, 23, 12, 5];
console.log(findMaxMinTS(numbers));
